### PR TITLE
EKS SQL MI scenario

### PR DESCRIPTION
**Pull request by dkirby-ms**

### CODE REVIEW

File: azure_arc_data_jumpstart/eks/mssql_mi/terraform/scripts/ClientTools.ps1 [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        Write-Host "Installing $app"
+        & choco install $app /y | Write-Output
+    }
+}
+
+# Downloading Azure Data Studio and azdata CLI
+Write-Host "Downloading Azure Data Studio and azdata CLI"
+Write-Host "`n"
+Invoke-WebRequest "https://azuredatastudio-update.azurewebsites.net/latest/win32-x64-archive/stable" -OutFile "C:\tmp\azuredatastudio.zip" | Out-Null
+Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/ekssqlmi/azure_arc_data_jumpstart/eks/mssql_mi/terraform/settings.json" -OutFile "C:\tmp\settings.json"
```
Comment (likamrat) [inline azure_arc_data_jumpstart/eks/mssql_mi/terraform/scripts/ClientTools.ps1:36]:
```suggestion
Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/main/azure_arc_data_jumpstart/eks/mssql_mi/terraform/settings.json" -OutFile "C:\tmp\settings.json"
```

File: azure_arc_data_jumpstart/eks/mssql_mi/terraform/scripts/ClientTools.ps1 [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+Invoke-WebRequest "$env:AZDATA_URL" -OutFile "C:\tmp\AZDataCLI.msi" | Out-Null
+Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/ekssqlmi/azure_arc_data_jumpstart/eks/mssql_mi/terraform/scripts/MSSQL_MI_Cleanup.ps1" -OutFile "C:\tmp\MSSQL_MI_Cleanup.ps1"
+Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/ekssqlmi/azure_arc_data_jumpstart/eks/mssql_mi/terraform/scripts/MSSQL_MI_Deploy.ps1" -OutFile "C:\tmp\MSSQL_MI_Deploy.ps1"
```
Comment (likamrat) [inline azure_arc_data_jumpstart/eks/mssql_mi/terraform/scripts/ClientTools.ps1:39]:
```suggestion
Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/main/azure_arc_data_jumpstart/eks/mssql_mi/terraform/scripts/MSSQL_MI_Cleanup.ps1" -OutFile "C:\tmp\MSSQL_MI_Cleanup.ps1"
Invoke-WebRequest "https://raw.githubusercontent.com/microsoft/azure_arc/main/azure_arc_data_jumpstart/eks/mssql_mi/terraform/scripts/MSSQL_MI_Deploy.ps1" -OutFile "C:\tmp\MSSQL_MI_Deploy.ps1"
```

File: azure_arc_data_jumpstart/eks/mssql_mi/terraform/scripts/MSSQL_MI_Deploy.ps1 [hunk 3]
```diff
@@ -0,0 +1,36 @@
+Start-Transcript -Path C:\tmp\mssql_mi_deploy.log
+
+# Deploying Azure Arc Data Controller and managed instance
+start PowerShell {for (0 -lt 1) {kubectl get pod -n $env:ARC_DC_NAME; sleep 5; clear }}
```
Comment (likamrat) [inline azure_arc_data_jumpstart/eks/mssql_mi/terraform/scripts/MSSQL_MI_Deploy.ps1:4]:
```suggestion
Start-Process PowerShell {for (0 -lt 1) {kubectl get pod -n $env:ARC_DC_NAME; Start-Sleep 5; Clear-Host }}
```

File: docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
+---
+type: docs
+title: "Data Controller Terraform plan"
+linkTitle: "Data Controller Terraform plan"
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md:4]: It's not Vanilla

File: docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+weight: 1
+description: >
+---
+
+## Deploy an Azure Arc Data Controller (Vanilla) on EKS using Terraform
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md:9]: It's not Vanilla

File: docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+
+  ```shell
+  az login
+  az ad sp create-for-rbac -n "<Unique SP Name>" --role contributor
+  ```
+
+  For example:
+
+  ```shell
+  az ad sp create-for-rbac -n "http://AzureArcK8s" --role contributor
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md:61]:
```suggestion
  az ad sp create-for-rbac -n "http://AzureArcData" --role contributor
```

File: docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+  ```
+
+  Output should look like this
+
+  ```json
+  {
+  "appId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+  "displayName": "AzureArcK8s",
+  "name": "http://AzureArcK8s",
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md:70]:
```suggestion
  "displayName": "AzureArcData",
  "name": "http://AzureArcData",
```

File: docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+
+* As part of the Windows Server 2019 VM deployment, there are 3 scripts executions:
+
+  1. *azure_arc.ps1* script will be created automatically as part of the Terraform plan runtime and is responsible on injecting the *TF_VAR* variables values on to the Windows instance which will then be used in both the *ClientTools* and the *LogonScript* scripts.
+
+  2. *ClientTools.ps1* script will run at the Terraform plan runtime Runtime and will:
+      * Create the *ClientTools.log* file  
+      * Install the required tools – az cli, az cli Powershell module, kubernetes-cli, aws-iam-authenticator, Visual C++ Redistributable (Chocolaty packages)
+      * Download Azure Data Studio & Azure Data CLI
+      * Download the *DC_Cleanup* and *DC_Deploy* Powershell scripts
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md:158]: Update script file names

File: docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+      * Open another Powershell session which will execute a command to watch the deployed Azure Arc Data Controller Kubernetes pods
+      * Deploy the Arc Data Controller using the *TF_VAR* variables values
+      * Deploy the MSSQL managed instance onto the data controller
+      * Restore a sample AdventureWorks database onto the data controller
+      * Unregister the logon script Windows schedule task so it will not run after first login
+      * Open Azure Data Studio
+
+## Deployment
+
+As mentioned, the Terraform plan will deploy an EKS cluster, the Azure Arc Data Controller on that cluster and an EC2 Windows Server 2019 Client instance.
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md:178]:
```suggestion
As mentioned, the Terraform plan will deploy an EKS cluster, the Azure Arc Data Controller on that cluster, a SQL Managed Instance with a sample database, and a Windows Server 2019 Client GCP compute instance.
```

File: docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+
+    ![PowerShell logon script run](./29.png)
+
+    ![PowerShell login script run](./30.png)
+
+    ![PowerShell login script run](./31.png)
+
+    ![PowerShell logon script run](./32.png)
+
+    ![Data Controller in a resource group](./33.png)
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md:273]: If you deployed in indirect mode, this should not be here.

File: docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md [hunk 11]
```diff
@@ -0,0 +1,312 @@
...
+
+* From Azure Data Studio, click on the MSSQL_MI instance and view the sample AdventureWorks database.
+
+  ![Sample AdventureWorks database](./36.png)
+
+## Cleanup
+
+* To delete the Azure Arc Data Controller and all of it's Kubernetes resources, run the *MSSQL_MI_Cleanup.ps1* PowerShell script located in *C:\tmp* on the Windows Client instance. At the end of it's run, the script will close all PowerShell sessions. **The Cleanup script run time is ~2-3min long**.
+
+  ![MSSQL_MI_Cleanup PowerShell script run](./38.png)
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md:296]: png jump 36<>38

File: docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md [hunk 12]
```diff
@@ -0,0 +1,312 @@
...
+
+    ![PowerShell login script run](./30.png)
+
+    ![PowerShell login script run](./31.png)
+
+    ![PowerShell logon script run](./32.png)
+
+    ![Data Controller in a resource group](./33.png)
+
+    ![Data Controller resource](./34.png)
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_data/eks/eks_mssql_mi/_index.md:275]: If you deployed in indirect mode, this should not be here.

**Review by likamrat — CHANGES_REQUESTED** [review]
- Scripts updates
- indirect mode scenario bug
- 36<>38 png jump

**Comment by dkirby-ms** [comment]
Requested changes made.

**Review by likamrat — APPROVED** [review]
LGTM

**Pull request merged**